src/context.c

```diff
@@ -290,6 +291,171 @@ ly_ctx_new(const char *search_dir, uint16_t options, struct ly_ctx **new_ctx)
     return rc;
 }
 
+static LY_ERR
+ly_ctx_new_yl_legacy(struct ly_ctx **ctx, struct lyd_node *yltree)
+{
+    struct lyd_node *module, *node;
+    struct ly_set *set;
+    const char **feature_arr;
+    LY_ERR ret;
+
+    LY_CHECK_RET(ret = lyd_find_xpath(yltree, "/ietf-yang-library:yang-library/modules-state/module", &set), ret);
+
+    /* process the data tree */
+    for (uint32_t i = 0; i < set->count; ++i) {
+        module = set->dnodes[i];
+
+        /* initiate */
+        const char *name = NULL, *revision = NULL;
+        struct ly_set features = {0};
+        ly_bool imported = 0;
```

Comment:
I am not sure about this. On one hand, the way you wrote it it is concise and clearly separate from the rest of the code. But at the same time variable declarations should always be at the beginning of a code block. I would prefer you always stick to this rule.